src/authorizations/users.rs

```diff
@@ -1,31 +1,76 @@
-use crate::config::Config;
+use reqwest::blocking::Client as reqwest_client;
+use serde::Deserialize;
 
-pub fn is_authorized(id: &str, config: &Config) -> bool {
-    config.authorized_users.contains(id)
+// Checks for permissions in https://github.com/rust-lang/team/
+#[derive(Debug, Deserialize, Clone)]
+struct TeamResponse {
+    discord_ids: Vec<usize>,
+}
+
+#[derive(Debug)]
+struct TeamClient {
+    client: reqwest_client,
+}
+
+impl TeamClient {
+    pub fn new() -> Self {
+        let team_client = reqwest_client::new();
+
+        TeamClient {
+            client: team_client,
+        }
+    }
+}
+
+pub fn is_authorized(id: &str) -> bool {
+    let authorization_info = team_info();
+    discord_id_in_list(id, authorization_info)
+}
+
+fn discord_id_in_list(id: &str, team_response: TeamResponse) -> bool {
+    team_response
+        .discord_ids
+        .contains(&id.parse::<usize>().unwrap())
+}
+
+fn team_info() -> TeamResponse {
+    get_team_info().unwrap()
+}
```

Comment:
Let's avoid unwrapping here. What I would do is change the `is_authorized` function to return a `Result<bool, Error>`, and wrap the error returned by `get_team_info` to add some context like "failed to fetch authorized users list from the Team API".